sycl/test/basic_tests/range_offset_fit_in_int.cpp

```diff
@@ -0,0 +1,177 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -D__SYCL_ID_QUERIES_FIT_IN_INT__=1 %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <climits>
+
+namespace S = cl::sycl;
+
+void checkRangeException(S::runtime_error &E) {
+  constexpr char Msg[] = "Provided range is out of integer limits. Suggest "
+                         "disabling `id-queries-fit-in-int32' optimizations "
+                         "flag.";
```

Comment:
What is `id-queries-fit-in-int32` optimizationS flag?
Does the compiler support such flag?
If it doesn't, please, suggest **working** solutions. E.g. using `__SYCL_ID_QUERIES_FIT_IN_INT__` macro.

#1685 refers to `fsycl-id-queries-fit-in-int`.

```suggestion
                         "disabling `id-queries-fit-in-int32' flag.";
```

Reply:
Yes, that's a typo.
Fixed.

Reply:
There are multiple typos in this string, which I think make it grammatically incorrect.
Also, I'd like to hear answers to my questions above.

Reply:
Done as discussed.